server/ngwmn/views.py

```diff
@@ -42,8 +45,53 @@ def version():
         'version': __version__
     })
 
+@app.route('/provider/', methods=['GET'])
+def providers():
+    """
+    NGWMN available providers view
+
+    """
+    return render_template('providers.html', providers=get_providers())
+
+
+@app.route('/provider/<agency_cd>/', methods=['GET'])
+def provider(agency_cd):
+    """
+    NGWMN provider information view
+    """
+    providers = get_providers()
+    providers_by_agency_cd = dict(map(lambda x: (x['agency_cd'], x), providers))
+    if agency_cd not in providers_by_agency_cd:
+        return '{0} is not a valid agency code'.format(agency_cd), 404
+
+    url = '{0}createrssfeed.action?types=page&spaces=GWDataPortal&title=X&labelString=ngwmn_provider_{1}_{2}&amp;excludedSpaceKeys%3D&sort=modified&maxResults=10&timeSpan=3600&showContent=true&confirm=Create+RSS+Feed'.format(
+        app.config['CONFLUENCE_URL'],
+        agency_cd,
+        'main'
+    )
+    return render_template('provider.html',
+                           agency_metadata=providers_by_agency_cd.get(agency_cd),
+                           provider_content=pull_feed(confluence_url(agency_cd, MAIN_CONTENT)),
+                           site_selection=pull_feed(confluence_url(agency_cd, SITE_SELECTION_CONTENT)),
+                           data_collection=pull_feed(confluence_url(agency_cd, DATA_COLLECTION_CONTENT)),
+                           data_management=pull_feed(confluence_url(agency_cd, DATA_MANAGEMENT_CONTENT)),
+                           other_agency_info=pull_feed(confluence_url(agency_cd, OTHER_AGENCY_INFO_CONTENT))
+                           )
```

Comment:
The more web requests like this we do, strengthens the case of switching to an async framework... we could speed these views up considerably, and handle more concurrency per container.

Reply:
I've researched a bit but haven't used any. There's also the aiohttp client library.